### ISSUE TITLE
Circular dependency on CGI::Application::PSGI in tests

### ISSUE THREAD

**Issue by rshingleton** [MartinMcGrath/CGI--Application#5]
On trying to install 4.60 we are seeing failures. 

On a new install it's impossible to get a clean install.

**CGI::Application fails giving the following:**
``` Can't locate CGI/Application/PSGI.pm in @INC (you may need to install the CGI::Application::PSGI module)```

Trying to install CGI::Application::PSGI (on a new perl install) fails due to dependency on CGI::Application.

On a perl install with an older version of CGI::Application, I can get CGI::Application::PSGI installed and then update CGI::Application.

I haven't tried to force the 4.60 on a fresh perl install yet.

**Comment by rshingleton** [comment]
Full log:
```
cpanm CGI::Application
--> Working on CGI::Application
Fetching http://www.cpan.org/authors/id/M/MA/MARTO/CGI-Application-4.60.tar.gz ... OK
Configuring CGI-Application-4.60 ... OK
Building and testing CGI-Application-4.60 ... FAIL
! Installing CGI::Application failed. See /Users/shingler/.cpanm/work/1519840236.96346/build.log for details. Retry with --force to force install it.
shingler-1mbpdu:dash_trunk shingler$ cat /Users/shingler/.cpanm/work/1519840236.96346/build.log
cpanm (App::cpanminus) 1.7042 on perl 5.025010 built for darwin-thread-multi-2level
Work directory is /Users/shingler/.cpanm/work/1519840236.96346
You have make /usr/bin/make
You have LWP 6.24
You have /usr/local/bin/tar: tar (GNU tar) 1.30
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching CGI::Application () on cpanmetadb ...
--> Working on CGI::Application
Fetching http://www.cpan.org/authors/id/M/MA/MARTO/CGI-Application-4.60.tar.gz
-> OK
Unpacking CGI-Application-4.60.tar.gz
Entering CGI-Application-4.60
Checking configure dependencies from META.json
Checking if you have ExtUtils::Install 1.46 ... Yes (2.04)
Checking if you have Module::Build 0.42 ... Yes (0.4220)
Configuring CGI-Application-4.60
Running Build.PL
Created MYMETA.yml and MYMETA.json
Creating new 'Build' script for 'CGI-Application' version '4.60'
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Test::More 0.47 ... Yes (1.302122)
Checking if you have Class::ISA 0 ... Yes (0.36)
Checking if you have Carp 0 ... Yes (1.42)
Checking if you have Module::Build 0 ... Yes (0.4220)
Checking if you have Test::Requires 0 ... Yes (0.10)
Checking if you have HTML::Template 0 ... Yes (2.95)
Checking if you have Scalar::Util 0 ... Yes (1.46_02)
Checking if you have CGI 4.21 ... Yes (4.38)
Building and testing CGI-Application-4.60
Building CGI-Application
t/arrayrefmodes.t ............ ok
t/basic.t .................... ok
t/callbacks.t ................ ok
t/copy.t ..................... ok
t/default_runmode.t .......... ok
t/enhancement31.t ............ ok
t/error_hook.t ............... ok
t/errormode.t ................ ok
t/false_runmode.t ............ ok
t/getquery.t ................. ok
t/header_props.t ............. ok
t/html_tmpl_class.t .......... ok
t/load_tmpl_default.t ........ ok
t/load_tmpl_hook.t ........... ok
t/mailform.t ................. ok
t/mode_param_overwritten.t ... ok
t/mode_param_path_info.t ..... ok
t/otherloadtmpl.t ............ ok
t/postrun.t .................. ok
t/prerun.t ................... ok
Can't locate CGI/Application/PSGI.pm in @INC (you may need to install the CGI::Application::PSGI module) (@INC contains: t/lib /Users/shingler/.cpanm/work/1519840236.96346/CGI-Application-4.60/blib/lib /Users/shingler/.cpanm/work/1519840236.96346/CGI-Application-4.60/blib/arch /prod/dash/shared/dependencies/perl5/perlbrew/perls/perl-5.25.10/lib/site_perl/5.25.10/darwin-thread-multi-2level /prod/dash/shared/dependencies/perl5/perlbrew/perls/perl-5.25.10/lib/site_perl/5.25.10 /prod/dash/shared/dependencies/perl5/perlbrew/perls/perl-5.25.10/lib/5.25.10/darwin-thread-multi-2level /prod/dash/shared/dependencies/perl5/perlbrew/perls/perl-5.25.10/lib/5.25.10 .) at t/psgi_streaming_callback.t line 8.
BEGIN failed--compilation aborted at t/psgi_streaming_callback.t line 8.
# Looks like your test exited with 2 before it could output anything.
t/psgi_streaming_callback.t ..
Dubious, test returned 2 (wstat 512, 0x200)
Failed 18/18 subtests
t/query.t .................... ok
t/run_as_psgi.t .............. ok
t/zerorm.t ................... ok

Test Summary Report
-------------------
t/psgi_streaming_callback.t (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 18 tests but ran 0.
Files=24, Tests=212,  2 wallclock secs ( 0.10 usr  0.06 sys +  1.98 cusr  0.34 csys =  2.48 CPU)
Result: FAIL
Failed 1/24 test programs. 0/212 subtests failed.
-> FAIL Installing CGI::Application failed. See /Users/shingler/.cpanm/work/1519840236.96346/build.log for details. Retry with --force to force install it.
Expiring 2 work directories.
shingler-1mbpdu:dash_trunk shingler$ cpanm CGI::Application::PSGI
--> Working on CGI::Application::PSGI
Fetching http://www.cpan.org/authors/id/M/MA/MARKSTOS/CGI-Application-PSGI-1.00.tar.gz ... OK
Configuring CGI-Application-PSGI-1.00 ... OK
Building and testing CGI-Application-PSGI-1.00 ... OK
Successfully installed CGI-Application-PSGI-1.00
1 distribution installed
shingler-1mbpdu:dash_trunk shingler$ cpanm CGI::Application
--> Working on CGI::Application
Fetching http://www.cpan.org/authors/id/M/MA/MARTO/CGI-Application-4.60.tar.gz ... OK
Configuring CGI-Application-4.60 ... OK
Building and testing CGI-Application-4.60 ... OK
Successfully installed CGI-Application-4.60 (upgraded from 4.50)
1 distribution installed
```

**Comment by MartinMcGrath** [comment]
Thanks for the report, I'll try and replicate this in the morning and do some digging.

**Comment by MartinMcGrath** [comment]
A quick test, fresh builds of 5.24.3  and 5.26.1 via perlbrew, and `cpanm -v CGI::Application` builds without issue. Let me run some more tests tomorrow

**Comment by eserte** [comment]
I am about to send 46 fail reports because of:
```
Can't locate CGI/Application/PSGI.pm in @INC (you may need to install the CGI::Application::PSGI module) (@INC contains: t/lib /home/cpansand/.cpan/build/2018022812/CGI-Application-4.60-4/blib/lib /home/cpansand/.cpan/build/2018022812/CGI-Application-4.60-4/blib/arch /usr/perl5.24.1p/lib/site_perl/5.24.1/amd64-freebsd /usr/perl5.24.1p/lib/site_perl/5.24.1 /usr/perl5.24.1p/lib/5.24.1/amd64-freebsd /usr/perl5.24.1p/lib/5.24.1 .) at t/psgi_streaming_callback.t line 8.
BEGIN failed--compilation aborted at t/psgi_streaming_callback.t line 8.
# Looks like your test exited with 2 before it could output anything.
t/psgi_streaming_callback.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 18/18 subtests 
```

**Comment by MaxPerl** [comment]
Dear Martin,
thank you very much for the new release!
I think the test script is wrong. We don't need any more CGI::Application::PSGI and the script - as far as I can see it - doesn't use it, too.

So I think, we should just change the following line:

`use CGI::Application::PSGI`

with

`use CGI::PSGI`

**Comment by MartinMcGrath** [comment]
@MaxPerl 

I think the issue may be to do with the MANIFEST file which was updated (in https://github.com/MartinMcGrath/CGI--Application/commit/3341b09cabe0e3a50d9db5db61ed0e777db8140b) to include things which look like they've either never been used, or not been used in a long time. 

On Debian using perlbrew I tested:

*  perl-5.24.3
*  perl-5.26.1
*  perl-5.27.9

On Windows 7 I tested with Strawberry v5.26.1.1 64bit.
Travis built and tested 8 different builds of perl (5.8 to 5.22).
None of these resulted in a problem/failure.

I'll fix the problem, let the tests re-run then if successful release again shortly after so there's a stable release. Once that's complete my plan is to tidy up the repository, alter the way I invoke travis to be more helpful in future, before addressing any of the other PRs that are waiting. 

I'll do some more digging as to why [this caused a problem in some placed but not others](http://matrix.cpantesters.org/?dist=CGI-Application+4.60), because I'd like to understand why exactly.
